nemo/collections/asr/data/audio_to_text_dali.py

```diff
@@ -320,22 +324,69 @@ def __init__(
             self.pad_value = params['pad_value'] if 'pad_value' in params else 0.0
 
         with self.pipe:
-            audio, indices = dali.fn.readers.nemo_asr(
-                name="Reader",
-                manifest_filepaths=manifest_filepath.split(','),
-                dtype=dali.types.FLOAT,
-                downmix=True,
-                sample_rate=float(self.sample_rate),
-                min_duration=min_duration,
-                max_duration=max_duration,
-                read_sample_rate=False,
-                read_text=False,
-                read_idxs=True,
-                random_shuffle=shuffle,
-                shard_id=self.shard_id,
-                num_shards=self.num_shards,
-                pad_last_batch=True,
-            )
+            if audio_tar_filepaths is None and audio_tar_index_filepaths is None:
+                audio, indices = dali.fn.readers.nemo_asr(
+                    name="Reader",
+                    manifest_filepaths=manifest_filepath.split(','),
+                    dtype=dali.types.FLOAT,
+                    downmix=True,
+                    sample_rate=float(self.sample_rate),
+                    min_duration=min_duration,
+                    max_duration=max_duration,
+                    read_sample_rate=False,
+                    read_text=False,
+                    read_idxs=True,
+                    random_shuffle=shuffle,
+                    shard_id=self.shard_id,
+                    num_shards=self.num_shards,
+                    pad_last_batch=True,
+                )
+
+                self.is_tarred_dataset = False
+
+            elif audio_tar_filepaths is not None and audio_tar_index_filepaths is not None:
+                audio_tar_filepaths = expand_audio_filepaths(
+                    audio_tar_filepaths, shard_strategy=shard_strategy, world_size=world_size, global_rank=global_rank
+                )
+                audio_tar_index_filepaths = expand_audio_filepaths(
+                    audio_tar_index_filepaths,
+                    shard_strategy=shard_strategy,
+                    world_size=world_size,
+                    global_rank=global_rank,
+                )
+
+                if len(audio_tar_filepaths) != len(audio_tar_index_filepaths):
+                    raise ValueError(
+                        f"Number of filepaths provided for `audio_tar_filepaths` must match "
+                        f"`audio_tar_index_filepaths`. Got {len(audio_tar_filepaths)} audio_tar_filepaths and "
```

Comment:
```suggestion
                        f"`audio_tar_index_filepaths` or f"`audio_tar_index_filepaths` should be empty. Got {len(audio_tar_filepaths)} audio_tar_filepaths and "
```

Reply:
Not selecting this change, as Nemo will enforce precomputed index for efficiency reasons on HPC. We don't want users to silently compute indices on the fly on millions of files in the tarred datasets.

Reply:
Sure. Up to you.